applications/GeoMechanicsApplication/custom_processes/apply_constant_boundary_phreatic_line_pressure_process.hpp

```diff
@@ -88,85 +85,51 @@ class ApplyConstantBoundaryPhreaticLinePressureProcess : public Process
-                else if (mIsFixedProvided) rNode.Free(var);
-
-                double height = 0.0;
-                if (rNode.Coordinates()[mHorizontalDirection] >= mMinHorizontalCoordinate && rNode.Coordinates()[mHorizontalDirection] <= mMaxHorizontalCoordinate) {
-                    height = mSlope * (rNode.Coordinates()[mHorizontalDirection] - mFirstReferenceCoordinate[mHorizontalDirection]) + mFirstReferenceCoordinate[mGravityDirection];
-                } else if (rNode.Coordinates()[mHorizontalDirection] < mMinHorizontalCoordinate) {
-                    height = mSlope * (mMinHorizontalCoordinate - mFirstReferenceCoordinate[mHorizontalDirection]) + mFirstReferenceCoordinate[mGravityDirection];
-                } else if (rNode.Coordinates()[mHorizontalDirection] > mMaxHorizontalCoordinate) {
-                    height = mSlope * (mMaxHorizontalCoordinate - mFirstReferenceCoordinate[mHorizontalDirection]) + mFirstReferenceCoordinate[mGravityDirection];
-                }
-                const double distance = height - rNode.Coordinates()[mGravityDirection];
-                const double pressure = mSpecificWeight * distance ;
-
-                if (pressure > 0.0) {
-                    rNode.FastGetSolutionStepValue(var) = pressure;
-                } else {
-                    rNode.FastGetSolutionStepValue(var) = 0.0;
-                }
-            });
-
-        }
+        const Variable<double> &var = KratosComponents< Variable<double> >::Get(mVariableName);
+
+        block_for_each(mrModelPart.Nodes(), [&var, this](Node& rNode){
+            if (mIsFixed) rNode.Fix(var);
+            else if (mIsFixedProvided) rNode.Free(var);
+
+            double horCoord = rNode.Coordinates()[mHorizontalDirection];
+            horCoord = std::max(horCoord, mMinHorizontalCoordinate);
+            horCoord = std::min(horCoord, mMaxHorizontalCoordinate);
+            const double height   = mSlope * (horCoord - mFirstReferenceCoordinate[mHorizontalDirection]) + mFirstReferenceCoordinate[mGravityDirection];
+            const double distance = height - rNode.Coordinates()[mGravityDirection];
+            const double pressure = mSpecificWeight * distance ;
+            rNode.FastGetSolutionStepValue(var) = std::max(pressure,0.0);
+        });
 
         KRATOS_CATCH("")
     }
 
+
     /// Turn back information as a string.
     std::string Info() const override
     {
         return "ApplyConstantBoundaryPhreaticLinePressureProcess";
     }
 
-    /// Print information about this object.
-    void PrintInfo(std::ostream& rOStream) const override
-    {
-        rOStream << "ApplyConstantBoundaryPhreaticLinePressureProcess";
-    }
-
-    /// Print object's data.
-    void PrintData(std::ostream& rOStream) const override
-    {
-    }
-
-///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-
 protected:
```

Comment:
I guess that at some later point in time we will make these data members `private`?

Reply:
Yes indeed.